src/main/java/org/sagebionetworks/bridge/fitbit/bridge/FitBitUserIterator.java

```diff
@@ -87,9 +95,17 @@ private FitBitUser getNextFitBitUser() {
         try {
             token = bridgeClientManager.getClient(ForWorkersApi.class).getOAuthAccessToken(studyId,
                     Constants.FITBIT_VENDOR_ID, healthCode).execute().body();
-        } catch (IOException ex) {
+        } catch (BridgeSDKException | IOException ex) {
+            // If it's a 4XX error, we know this is a deterministic error. Don't try again. Advance the nextIndex.
+            if (ex instanceof BridgeSDKException) {
+                int statusCode = ((BridgeSDKException) ex).getStatusCode();
+                if (statusCode >= 400 && statusCode <= 499) {
+                    nextIndex++;
+                }
+            }
```

Comment:
Is there any situation where you don't want to advance if the alternative is looping over the same page until you don't get an error... for forever?

Reply:
If it's a non-deterministic error, like a 500 Internal Server error, the argument can be made that the iterator should retry.

However, a counter-argument can be made that the iterator shouldn't have this logic, and that the retry logic should be in a lower layer, and that the iterator should always advance the iterator. See https://sagebionetworks.jira.com/browse/BRIDGE-2055